### DISCUSSION TITLE
fitHMM in parallel

### DISCUSSION THREAD

**Discussion by riccile** [bmcclintock/momentuHMM#75] [Q&A] [answered]
Hi,
I am trying to run a model with four covariates (formula ~ a * b * c * d) but I'm running into trouble with the computational time needed. Do you have any suggestions for fitting the HMM in parallel or other ways to speed up the process?
Thanks!

**Comment by bmcclintock** [comment, marked as answer]
To my knowledge there is no way to "fit the HMM in parallel".  Run times are a function of computing power, the length of the times series, the complexity of the model (e.g. number of states and/or covariates), and how "good" the initial values are for the optimization. Not much can be done about the first two, but see https://github.com/bmcclintock/momentuHMM/issues/22#issuecomment-519603140 for tips on the latter two. Also note that a 4-way interaction results in `nbStates*(nbStates-1)*16` covariates -- that's a lot!
